### ISSUE TITLE
alicloud_vswitch custom delete timeout is not respected

### ISSUE THREAD

**Issue by ialidzhikov** [aliyun/terraform-provider-alicloud#2734]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

- terraform version
   ```
   $ terraform -v
   Terraform v0.12.20
    ```

- terraform-provider-alicloud version - `1.84.0`

### Affected Resource(s)
Please list the resources as a list, for example:
- alicloud_vswitch

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
provider "alicloud" {
  access_key = var.ACCESS_KEY_ID
  secret_key = var.ACCESS_KEY_SECRET
  region = "eu-central-1"
}

resource "alicloud_vpc" "vpc" {
  name       = "foo"
  cidr_block = "10.250.0.0/16"
}

resource "alicloud_vswitch" "vsw_z0" {
  name              = "bar"
  vpc_id            = alicloud_vpc.vpc.id
  cidr_block        = "10.250.0.0/19"
  availability_zone = "eu-central-1b"

  timeouts {
    create = "5m"
    delete = "5m"
  }
}
```

### Debug Output
Please provider a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
What should have happened?

I expect the configured timeout for the `alicloud_vswitch` deletion to be respected and the `alicloud_vswitch` deletion to time out with appropriate error after `5m`.

### Actual Behavior
What actually happened?

`alicloud_vswitch` deletion times out after `20m`, not `5m`.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

2. Manually add dependency to the vswitch to be able to reproduce the issue with the deletion timeout.

3. `terraform destroy`

```
$ terraform destroy
Initializing the backend...

Initializing provider plugins...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.alicloud: version = "~> 1.84"
* provider.null: version = "~> 2.1"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
alicloud_vpc.vpc: Refreshing state... [id=vpc-1234]
alicloud_vswitch.vsw_z0: Refreshing state... [id=vsw-1234]
alicloud_vswitch.vsw_z0: Destroying... [id=vsw-1234]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 1m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 1m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 1m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 1m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 1m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 1m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 2m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 2m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 2m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 2m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 2m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 2m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 3m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 3m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 3m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 3m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 3m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 3m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 4m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 4m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 4m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 4m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 4m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 4m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 5m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 5m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 5m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 5m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 5m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 5m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 6m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 6m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 6m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 6m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 6m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 6m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 7m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 7m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 7m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 7m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 7m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 7m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 8m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 8m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 8m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 8m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 8m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 8m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 9m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 9m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 9m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 9m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 9m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 9m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 10m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 11m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 11m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 11m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 11m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 11m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 11m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 12m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 12m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 12m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 12m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 12m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 12m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 13m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 13m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 13m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 13m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 13m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 13m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 14m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 14m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 14m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 14m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 14m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 14m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 15m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 15m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 15m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 15m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 15m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 15m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 16m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 16m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 16m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 16m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 16m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 16m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 17m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 17m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 17m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 17m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 17m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 17m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 18m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 18m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 18m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 18m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 18m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 18m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 19m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 19m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 19m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 19m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 19m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 19m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-1234, 20m0s elapsed]
alicloud_vpc.vpc: Destroying... [id=vpc-1234]

Destroy complete! Resources: 0 destroyed.
```

### Important Factoids
Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

**Comment by ialidzhikov** [comment]
When I check the state, I see that the `timeouts` is `null`:
```json
        {
          "mode": "managed",
          "type": "alicloud_vswitch",
          "name": "vsw_z0",
          "provider": "provider.alicloud",
          "instances": [
            {
              "schema_version": 0,
              "attributes": {
                "availability_zone": "eu-central-1b",
                "cidr_block": "10.250.0.0/19",
                "description": "",
                "id": "vsw-1234",
                "name": "bar",
                "tags": {},
                "timeouts": null,
                "vpc_id": "vpc-1234"
              },
              "private": "foo",
              "dependencies": [
                "alicloud_vpc.vpc"
              ]
            }
          ]
        },
```

This seems to be related to https://github.com/hashicorp/terraform/issues/22810 and https://github.com/hashicorp/terraform/issues/21528.

@jbardin, @apparentlymart can you help with this issue?

It seems to me that the issue is in `github.com/hashicorp/terraform-plugin-sdk`. Currently terraform-provider-alicloud is vendoring `github.com/hashicorp/terraform-plugin-sdk@v1.4.0`.

**Comment by ialidzhikov** [comment]
I also see that for newly applied infrastructure, the timeouts field in the vswitch state has expected values:
```json
        {
          "mode": "managed",
          "type": "alicloud_vswitch",
          "name": "vsw_z0",
          "provider": "provider.alicloud",
          "instances": [
            {
              "schema_version": 0,
              "attributes": {
                "availability_zone": "eu-central-1a",
                "cidr_block": "10.250.0.0/17",
                "description": "",
                "id": "vsw-5678",
                "name": "baz",
                "tags": {},
                "timeouts": {
                  "create": "5m",
                  "delete": "5m"
                },
                "vpc_id": "vpc-5678"
              }
            }
          ]
        }
```

But for infrastructure which is created since longtime, the timeouts are null as showed in the example state above.

**Comment by ialidzhikov** [comment]
I can confirm that there is not effect even when timeouts is non-null:

For the vswitch from above (`vsw-5678`) with timeouts `{"create": "5m","delete": "5m"}` in the terraform.tfstate:

```
$ terraform destroy

Initializing the backend...

Initializing provider plugins...

The following providers do not have any version constraints in configuration,
so the latest version was installed.

To prevent automatic upgrades to new major versions that may contain breaking
changes, it is recommended to add version = "..." constraints to the
corresponding provider blocks in configuration, with the constraint strings
suggested below.

* provider.alicloud: version = "~> 1.84"
* provider.null: version = "~> 2.1"

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
alicloud_vpc.vpc: Refreshing state... [id=vpc-5678]
alicloud_vswitch.vsw_z0: Refreshing state... [id=vsw-5678]
alicloud_vswitch.vsw_z0: Destroying... [id=vsw-5678]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 1m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 1m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 1m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 1m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 1m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 1m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 2m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 2m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 2m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 2m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 2m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 2m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 3m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 3m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 3m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 3m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 3m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 3m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 4m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 4m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 4m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 4m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 4m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 4m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 5m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 5m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 5m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 5m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 5m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 5m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 6m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 6m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 6m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 6m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 6m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 6m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 7m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 7m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 7m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 7m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 7m40s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 7m50s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 8m0s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 8m10s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 8m20s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 8m30s elapsed]
alicloud_vswitch.vsw_z0: Still destroying... [id=vsw-5678, 8m40s elapsed]

# full output omitted
```

**Comment by ialidzhikov** [comment]
ping @xiaozhu36, @zql-tqy, @zhuzhih2017, @Lexsss

**Comment by chushenmeshile** [comment]
@ialidzhikov  I recreated the resource and didn't duplicate your problem. I created and deleted it soon

**Comment by ialidzhikov** [comment]
@chushenmeshile , here are more clear steps to reproduce:

```hcl
provider "alicloud" {
  access_key = var.ACCESS_KEY_ID
  secret_key = var.ACCESS_KEY_SECRET
  region = "eu-central-1"
}

resource "alicloud_vpc" "vpc" {
  name       = "foo"
  cidr_block = "10.250.0.0/16"
}

resource "alicloud_vswitch" "vsw_z0" {
  name              = "bar"
  vpc_id            = alicloud_vpc.vpc.id
  cidr_block        = "10.250.0.0/19"
  availability_zone = "eu-central-1b"

  timeouts {
    create = "5m"
    delete = "5m"
  }
}

```


1. `terraform apply`

2. Create an alicloud instance with the newly created vswitch - our idea is to create a dependency that blocks the deletion of the vsiwtch

3. Run `terraform destroy` and ensure that the destroy fails with `DependencyViolation` error after the configured delete timeout (`5m`) -> **the custom delete timeout is respected**

4. Run `terraform destroy` again and ensure that the destroy fails with `DependencyViolation` after the default delete timeout (`20m`) -> **the custom delete timeout is NOT respected**


I experience the same bug also with other providers - see https://github.com/hashicorp/terraform-provider-aws/issues/16910.

**Comment by ialidzhikov** [comment]
It was an upstream terraform issue (hence affecting all terraform providers) - https://github.com/hashicorp/terraform/issues/27452 that is fixed in 0.15.
